src/main.js

```diff
@@ -298,108 +298,38 @@ let destinationsList = [
+    galleryList.classList.remove('visited');
   });
-  bucketlistCaptionsList.forEach(item =>  {
-    item.nextElementSibling.classList.remove('active');
-  });
-})
-
 
-bucketlistButton.addEventListener('click', () => {
-  //get two lists of images with class .visited and .bucketlist
-  const visitedImgList= document.querySelectorAll('.visited');
-  const bucketlistImgList= document.querySelectorAll('.bucketlist');
-
-  //get two lists of captions that are sibilings to images with class .visited and .bucketlist
-  const visitedCaptionsList= document.querySelectorAll('.visited');
-  const bucketlistCaptionsList= document.querySelectorAll('.bucketlist');
-
-  //remove .active from .visited images and add to .bucketlist
-  visitedImgList.forEach(item =>  {
-    item.classList.remove('active')
-  });
-  bucketlistImgList.forEach(item =>  {
-    item.classList.add('active')
+  allButton.addEventListener('click', () => {
+    galleryList.classList.remove('bucketlist');
+    galleryList.classList.remove('visited');
   });
 
-  //remove .active from .visited captions and add to .bucketlist
-  visitedCaptionsList.forEach(item =>  {
-    item.nextElementSibling.classList.remove('active');
-  });
-  bucketlistCaptionsList.forEach(item =>  {
-    item.nextElementSibling.classList.add('active');
-  });
-
-});
-
-
-allButton.addEventListener('click', () => {
-  //get two lists of images with class .visited and .bucketlist
-  const visitedImgList= document.querySelectorAll('.visited');
-  const bucketlistImgList= document.querySelectorAll('.bucketlist');
-
-  //get two lists of captions that are sibilings to images with class .visited and .bucketlist
-  const visitedCaptionsList= document.querySelectorAll('.visited');
-  const bucketlistCaptionsList= document.querySelectorAll('.bucketlist');
-
-  //add .active to all 
-  visitedImgList.forEach(item =>  {
-    item.classList.add('active')
-  });
-  bucketlistImgList.forEach(item =>  {
-    item.classList.add('active')
-  });
-  visitedCaptionsList.forEach(item =>  {
-    item.nextElementSibling.classList.add('active');
-  });
-  bucketlistCaptionsList.forEach(item =>  {
-    item.nextElementSibling.classList.add('active');
-  });
-
-});
-
-
-const buttonIsActive = () => {
-  const buttons= document.querySelectorAll('.button');
-  buttons.forEach(button => {
-      button.addEventListener('click', clicked => {
-          clicked.currentTarget.classList.toggle('active-button');
-      })
-  });
-}
-buttonIsActive();
-
-// const buttonRemoveActive = () => {
-//   if(buttonIsActive()) {
-//                 clicked.currentTarget.classList.remove('active-button');
-//     };
-// }
-// buttonRemoveActive();
+  //get clicked button
+  const buttonIsActive = () => {
+    const buttons= document.querySelectorAll('.button');
+    buttons.forEach(button => {
+        button.addEventListener('click', clicked => {
+            clicked.currentTarget.classList.add('active-button');
```

Comment:
When another btn is clicked, it should remove this class from the other buttons. Otherwise all bts will have `.active-button`.

Reply:
I was just about to ask you for a hint about that :D is there a way to get the previous button and remove .active from it?

Reply:
I tried toggle and add/remove but didn't get it right

Reply:
You'll need to loop through the buttons inside the click eventlistener. Loop through them and remove the class. Then target the current one and add the list.

Tip: It's a common thing to call event listener parameter as `event`, because we receive info about that event. calling it `clicked` might be confusing/misleading.

Reply:
>  Loop through them and remove the class. Then target the current one and add the list.

I'm not getting the order of this. You mean remove the active class?

Reply:
Here's a pseudo-code to help you:

```
// add click event to each button
//      loop through each button
//              remove the active class
//      add active class to the current button
``

Reply:
But why remove it and _then_ add? Why loop once to add click to each button and then again through each button?

Reply:
I might need to get back to this one later, I feel like I'm missing something fundamental here

Reply:
This is the closest I got, with some unexpected results 😄 

`buttons.forEach(button => {
  button.addEventListener('click', event => {
    current = event.currentTarget;
    for(let i=0; i<buttons.length && button!==current; i++) {

        button.classList.remove('active-button');
    }
    button = event.currentTarget.classList.add('active-button');
  });
});`

Reply:
So a break helped, I think I got it right 😄 Not a 100% sure why exactly it works, but it works

Reply:
Yey, you made it worked indeed! One cool thing about JS is that there are a lot of different ways to approach something! It wasn't exactly as I suggested but the goal is completed and that's what matters now.


After a break, I also had a better idea to approach this without needing to use a loop (which indeed is too much overload). The idea is to store a variable `elNavBtnActive` that contains the current clicked button. Then, in the btn click event, use that variable to remove the active class and replace the variable with the recent clicked nav button. That way we don't need to loop through the buttons to remove the active class.

Try to come up with a solution (really try). If after 10min you didn't figure it out, here's the solution down below:

<details>
<summary>Try by yourself before seeing the solution here</summary>

<details>
<summary>I hope you really did try! Here it goes..</summary>

```js
const buttons = document.querySelectorAll(".button");
let elButtonActive = null; // initially there isn't any active/clicked button, so this is null

buttons.forEach((button) => {
  button.addEventListener("click", (event) => {
    const clickedButton = event.target;
    
    // If there's a current actibe button, remove the class
    if (elButtonActive) elButtonActive.classList.remove("active-button");

    // Add active class to the current clicked button
    clickedButton.classList.add("active-button");

    // Replace the variable with the most recent (current) clicked button
    elButtonActive = clickedButton;
  });
});
```


</details>

</details>

Reply:
Here's what didn't work, at least it was in somewhat of a good direction. It's great having something to compare with as in most cases when I get something wrong it's hard to tell why it's that way and how to fix it. Your example here gives me something to learn from 

```
buttons.forEach(button => {
  button.addEventListener('click', event => {
  const elNavBtnActive = event.currentTarget;

  if(!elNavBtnActive) {
    elNavBtnActive=elNavBtnActive.className.replace(" active-button", "");
  }
  elNavBtnActive.classList.add('active-button')
});
});
```

Reply:
So you got it almost, the problem was the scope of `elNavBtnActive`. It was created inside the event listener, so each time the click happens, a new variable is created, forgetting the value stored in the previous click. That's why `elNavBtnActive` must be created in an outside scope (as in my example).

Does it make sense? I recommend you to review the concept of scoping in JS :)

Reply:
It does! The learn js course has basically nothing on scope so I very vaguely understand it, I'll have to look up another source